### PR TITLE
[TwigBridge] remove deprecated features

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Form/TwigRenderer.php [hunk 1]
```diff
@@ -17,25 +17,14 @@
...
-class TwigRenderer extends FormRenderer implements TwigRendererInterface
+class TwigRenderer extends FormRenderer
 {
-    /**
-     * @var TwigRendererEngineInterface
-     */
     private $engine;
 
-    public function __construct(TwigRendererEngineInterface $engine, CsrfTokenManagerInterface $csrfTokenManager = null)
+    public function __construct(TwigRendererEngine $engine, CsrfTokenManagerInterface $csrfTokenManager = null)
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Form/TwigRenderer.php]: shouldn't the typeint be in FormRendererEngineInterface ?

File: src/Symfony/Bridge/Twig/Form/TwigRenderer.php [hunk 2]
```diff
@@ -17,25 +17,14 @@
...
     {
         parent::__construct($engine, $csrfTokenManager);
 
         $this->engine = $engine;
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Form/TwigRenderer.php]: you don't actually need this property, as it is now unused.

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Bridge/Twig/Form/TwigRenderer.php (hunk 1 above)
Comment (xabbuh) [inline src/Symfony/Bridge/Twig/Form/TwigRenderer.php]: Not sure, shouldn't the Twig extension be sure that the underlying renderer engine supports Twig? By the way, if we did that change, that would mean that we could entirely drop the `TwigRenderer` class and use the parent `AbstractRendererEngine` class instead.

File: src/Symfony/Bridge/Twig/Form/TwigRenderer.php [hunk 3]
```diff
@@ -17,25 +17,10 @@
 /**
  * @author Bernhard Schussek <bschussek@gmail.com>
  */
-class TwigRenderer extends FormRenderer implements TwigRendererInterface
+class TwigRenderer extends FormRenderer
```
Comment (Tobion) [inline src/Symfony/Bridge/Twig/Form/TwigRenderer.php]: This class has no value anymore. `Symfony\Component\Form\FormRenderer` can be used directly now. So TwigRenderer  should be deprecated and removed IMO.

**Review by Tobion — CHANGES_REQUESTED** [review]

File: src/Symfony/Bridge/Twig/Form/TwigRenderer.php (hunk 3 above)
Comment (Tobion) [inline src/Symfony/Bridge/Twig/Form/TwigRenderer.php]: https://github.com/symfony/symfony/pull/23437

**Comment by nicolas-grekas** [comment]
@xabbuh would be great to finish this one, that's one of the two last PR removing deprecated features. Thanks!

**Comment by xabbuh** [comment]
@nicolas-grekas I think we should wait for #23437 to be able to remove the `TwigRenderer` class too.

**Comment by fabpot** [comment]
#23437 merged now :)

**Comment by nicolas-grekas** [comment]
rebase unlocked :)

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**